usecases/config/config_handler.go

```diff
@@ -91,8 +92,7 @@ type Flags struct {
 }
 
 type SchemaHandlerConfig struct {
-	MaximumAllowedCollectionsCount   int         `json:"maximum_allowed_collections_count" yaml:"maximum_allowed_collections_count"`
-	MaximumAllowedCollectionsCountFn func() *int `json:"-" yaml:"-"`
+	MaximumAllowedCollectionsCount runtime.DynamicValue[int] `json:"maximum_allowed_collections_count" yaml:"maximum_allowed_collections_count"`
```

Comment:
I would rather use a pointer to a DynamicValue, as it is very easy to introduce a subtle bug by copying the atomic value, which then would not be updated by the updater.

Example test:
```go
func TestAccidentallyCopyingAtomicValue(t *testing.T) {
	a := NewFlag[int](1)

	require.Equal(t, 1, a.val.Load())

	// copy the flag
	b := a

	// update the flag with new value
	a.val.Store(10)

	// expect the update to be propagated
	require.Equal(t, 10, b.val.Load()) // fails :(
}

```

Reply:
Good point. My only concern was then using `cfg = WeaviateConfig{}` it in tests will break because these pointer types would be `nil` calling `val.Get()` panic in tests.

But found a way to workaround that, By handling the `nil` case in `Get()` and returning correct zero value. Locked with some tests.